src/window.cpp

```diff
@@ -3061,38 +3061,44 @@ void InputLoop()
 	HandleAutoscroll();
 }
 
+static const uint TICK_15_INTERVAL = 15 * MILLISECONDS_PER_TICK;
+static const uint TICK_100_INTERVAL = 100 * MILLISECONDS_PER_TICK;
+
 /**
  * Update the continuously changing contents of the windows, such as the viewports
  */
 void UpdateWindows()
 {
 	Window *w;
 
-	static int highlight_timer = 1;
-	if (--highlight_timer == 0) {
-		highlight_timer = 15;
+	static uint32 next_highlight_time = _realtime_tick + TICK_15_INTERVAL;
+	if (_realtime_tick >= next_highlight_time)
+	{
+		next_highlight_time = _realtime_tick + TICK_15_INTERVAL;
 		_window_highlight_colour = !_window_highlight_colour;
 	}
 
 	FOR_ALL_WINDOWS_FROM_FRONT(w) {
 		w->ProcessScheduledInvalidations();
 		w->ProcessHighlightedInvalidations();
 	}
+}
+
+void DrawWindows()
+{
+	Window *w;
 
 	/* Skip the actual drawing on dedicated servers without screen.
 	 * But still empty the invalidation queues above. */
```

Comment:
Comment is not invalid; but is this line also needed? (I assume it is not called from a dedicated flow?)